util/histogram-median/main.c

```diff
@@ -52,15 +52,32 @@ int main(int argc, char **argv)
 
   info("read histogram from %s\n", path);
 
-  double value;
-
-  if (median(hist, &value) != 0)
+  if (histogram_num_bins(hist) == 0)
     {
-      fprintf(stderr, "error calculating median\n");
-      return EXIT_FAILURE;
+      if (opt.empty_value_given)
+        {
+          info("median value: ");
+          printf("%s\n", opt.empty_value_arg);
```

Comment:
So the `opt.empty_value_arg` is a string, and it might be `0.0` or `NaN`, depends on what the caller would like.  But `value` in line 79 is a double, so the `%g` will print `0.00312343` or `3.312343e-03` or whatever.  The point here is that the it is the caller who needs to handle the empty-histogram case, they might have `EMPTY` as the empty value, then they read the output of the program, if it is `EMPTY` then they handle that case, else they treat the value as a float

Reply:
Got you! And that's why in the SR we added the NaN check that was passed as empty-value! :+1:

Reply:
for info, `%g` is for double precision argument, and it formats like `%e` or `%f` depending on the size of the number, so 0.00000003 would be printed as 3e-09, but 3.00000 would be printed as 3.0, you add extra numbers in there to the maximal width and number of places, so you see things like `%10.6g`, see `man 3 printf` for the gory details

Reply:
That's it, unfortunately the Ruby `sscanf` doesn't understand the `NaN` string (the C one does), so we have to handle it ourselves ...